neo4j/internal/bolt/dechunker_test.go

```diff
@@ -69,19 +70,26 @@ func TestDechunker(t *testing.T) {
 			}
 			str.Write(buf)
 		}
-		// Write end of mesage marker
+		// Write end of message marker
 		str.Write([]byte{0x00, 0x00})
 
 		// Dechunk the message
 		var msgBuf []byte
-		buf, msgBuf, err = dechunkMessage(str, buf)
+		serv, cli := net.Pipe()
+		go func() {
+			_, err := cli.Write(str.Bytes())
+			AssertNoError(t, err)
+		}()
+		buf, msgBuf, err = dechunkMessage(serv, buf, -1, nil, "")
 		AssertNoError(t, err)
 		AssertLen(t, msgBuf, int(msg.size))
 		// Check content of buffer
 		b = 0
 		for i := range msgBuf {
 			if msgBuf[i] != b {
 				t.Errorf("Wrong content in buffer at %d, %d vs %d (message %d)", i, msgBuf[i], b, msgi)
+				AssertNoError(t, serv.Close())
```

Comment:
Is that a way to unit test the `dechunker` reset timeout logic?

Reply:
I only relied on the existing Testkit tests. Is it worth adding a test here as well in your opinion?

Reply:
And now there is an integration test as well